marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -189,19 +217,19 @@ marlowePlutusContract = do
                 case wr of
                     ContractEnded -> do
                         logInfo @String $ "Contract Ended"
-                        pure ()
+                        marlowePlutusContract
                     Timeout _ -> do
                         logInfo @String $ "Contract Timeout"
                         continueWith marloweData
                     WaitingResult marloweData -> continueWith marloweData
 
             CloseContract -> do
                 logInfo @String $ "CloseContract"
-                pure ()
+                marlowePlutusContract
 
             NotSure -> do
                 logInfo @String $ "NotSure"
-                apply `select` wait `select` auto
+                marlowePlutusContract
```

Comment:
Previously, I executed the contract until it's Done (`pure()`).
To allow `redeem` we need a contract running after it is `Close`.
That's why I changed assertions in tests to `assertNotDone`.
That's why the contract is executed forever, and can be reused for multiple contracts.

Reply:
Ah, I see so there are two things we may want to consider, not sure whether they apply:
- Does this translate in higher fees? Does the person redeeming have to pay more because the transaction requires a larger validator?
- Will we have the fan-in fan-out problem? If the contract is reused, will the participants from different contract have to fight to get the transaction on the chain?

Reply:
In any case, this doesn't apply to the test, just worth thinking about it for the actual Wallet later

Reply:
This is the off-chain part of the contract, so it doesn't impact fees at all, and it won't affect on-chain validator size as well.
Similarly, a wallet can instantiate as many of these contracts as one wishes and run those in parallel.
Should be fine.